internal/datastore/proxy/caching_test.go

```diff
@@ -25,6 +26,19 @@ const (
 	nsB = "namespace_b"
 )
 
+// TestNilUnmarshal asserts that if we get a nil NamespaceDefinition from a
+// datastore implementation, the process of inserting it into the cache and
+// back does not break anything.
+func TestNilUnmarshal(t *testing.T) {
+	nsDef := (*core.NamespaceDefinition)(nil)
+	marshalled, err := nsDef.MarshalVT()
+	require.Nil(t, err)
+	var newDef *core.NamespaceDefinition
```

Comment:
newline before this line

Might also be worth adding another test for a "not found" entry, if we don't have one already

Reply:
Not sure how to go about that with the least setup. I'm willing to add it if you can point me in the right direction.

Reply:
Let's do it as a follow up